plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/names/KotlinName.kt

```diff
@@ -25,7 +26,11 @@ abstract class PrefixedKotlinName(prefix: String, name: Name) : KotlinName {
     override val mangled: String = "${prefix}_${name.asStringStripSpecialMarkers()}"
 }
 
-data class FunctionKotlinName(val name: Name) : PrefixedKotlinName("fun", name)
+abstract class PrefixedKotlinNameWithType(prefix: String, name: Name, type: TypeEmbedding) : KotlinName {
+    override val mangled: String = "${prefix}_${name.asStringStripSpecialMarkers()}${'$'}${type.name.mangled}"
```

Comment:
Arguably `"\$"` instead of `"{'$'}"` is more readable.

Reply:
Oh right, it was only raw string literals where that doesn't work.  Thanks, fixed!